packages/@aws-cdk/aws-cloudformation-codepipeline/lib/pipeline-action.ts

```diff
@@ -0,0 +1,318 @@
+import codepipeline  = require('@aws-cdk/aws-codepipeline');
+import iam = require('@aws-cdk/aws-iam');
+import cdk = require('@aws-cdk/cdk');
+
+// tslint:disable:max-line-length
+/**
+ * Properties common to all CloudFormation actions
+ */
+export interface CloudFormationCommonProps {
+    /**
+     * The name of the stack to apply this action to
+     */
+    stackName: string;
+
+    /**
+     * A name for the filename in the output artifact to store the AWS CloudFormation call's result.
+     *
+     * The file will contain the result of the call to AWS CloudFormation (for example
+     * the call to UpdateStack or CreateChangeSet).
+     *
+     * AWS CodePipeline adds the file to the output artifact after performing
+     * the specified action.
+     *
+     * @default No output artifact generated
+     */
+    outputFileName?: string;
+
+    /**
+     * The name of the output artifact to generate
+     *
+     * Only applied if `outputFileName` is set as well.
```

Comment:
I'd be tempted to have
```
output?: {
    fileName: string;
    artifactName?: string;
}
```

So you cannot specify `outputArtifactName` if it makes no sense.

Reply:
Mmmyeah. But in general we keep the property namespace flat.

Reply:
That's why I only say *tempted*.

Reply:
Agree with Rico.Optimize for the common case which is to just specify the file name

Reply:
Generally artifact names exist only to allow references in the Pipeline model which are achieved much better with object references in the CDK.

Copy: @skinny89